### ISSUE TITLE
I have a question about phrase.

### ISSUE THREAD

**Issue by DopeorNope-Lee** [Gryphe/MergeMonster#3]
```python

# Phrase = What to measure, weight = multiplication factor, contexts = proceeding contexts
bad_phrases:
  - phrase: "anticipation"
    weight: 10
    contexts: ["Her body quivers with ", "The atmosphere is thick with "]
  - phrase: "unwavering"
    weight: 1
    contexts: ["Filled with an "]
  - phrase: "determination"
    weight: 1
    contexts: ["Her eyes were filled with "]

# Note - Example of a complex phrase
good_phrases:
  - phrase: "The apple is in the bedroom"
    weight: 1
    contexts: ["Question: If I'm in the living room and pick up the apple, go to the bedroom and drop the apple, then walk to the kitchen, where is the apple? Explain your reasoning. Answer: "]

```
    
    
I saw a config_yaml in your base repo.

However, I don't know what the phrase means here. I'm going to merge the models of Lamma 2, but if I know the meaning of this phase, I think I can use it well.

Best,

**Comment by Gryphe** [comment]
That all depends on which phrases you're referring to here. In the case of good_phrases, the example is a correct answer to a question/riddle, formatted like you would normally talk to a language model. This example is a classic which a lot of models tend to answer wrong.

That example uses "Question:"  and "Answer:", but you could also use "User:" and "Assistant:". It's a language model, so any text goes!

Phrases could also be called auto-completions - MergeMonster tries to either decrease/increase (bad/good) the chance of text being auto-completed by selectively taking parts from models that it considers to be beneficial.

**Comment by DopeorNope-Lee** [comment]
@Gryphe Thank you for your kindness reply. I understand it as good_phrase means our preferred output from our merged model, and bad phrase means our unpreferred output from our merged model.

It seems like DPO method or give a loss to our merged model using KL-divergence. 

Am I right to understand like this?

**Comment by Gryphe** [comment]
You're 100% correct! In many ways it's DPO on a budget.

I'll consider the "issue" closed for now, but feel free to reopen it if you find yourself having more questions.

**Comment by choprahetarth** [comment]
I have a follow-up question. 
For example, if I have two models that do not do well on let's say 100 examples. 
Can something like this be done -> the bad phrases be those 100 examples?
Something like XGBoost, except instead of increasing the weight of the bad samples, you try to overfit the merged models on the given samples (which might be a good thing for the overall model?)

**Comment by Gryphe** [comment]
That should be a possibility, yes! "Phrase" is a simple text field that opens a whole world of possibilities, many of which I've yet to discover myself. 

By adding a long list of "bad examples" you could encourage the production of merges that are less likely to produce similar examples.
